api.go

```diff
@@ -571,11 +583,36 @@ func (res *resource) handleUpdate(w http.ResponseWriter, r *http.Request, ps htt
 
 	updatingObj := updatingObjs.Index(0).Interface()
 
-	if err := res.source.Update(updatingObj, buildRequest(r)); err != nil {
+	response, err := res.source.Update(updatingObj, buildRequest(r))
+
+	if err != nil {
 		return err
 	}
-	w.WriteHeader(http.StatusNoContent)
-	return nil
+
+	switch response.StatusCode() {
```

Comment:
can we move those three switch response.StatusCode() statements into one function?